### ISSUE TITLE
Spec should specify what the changes trie is

### ISSUE THREAD

**Issue by tomaka** [w3f/polkadot-spec#215]
Section E.1.8 specifies the `ext_storage_changes_root` function. This function is actually supposed to return the root of what is called the changes trie.

The "changes tries" is a trie maintained in parallel with the storage trie and is briefly explained in Substrate's code here: https://github.com/paritytech/substrate/blob/dbf2163250833e6ac898e7f6c3c8f89f08a7c19d/primitives/state-machine/src/changes_trie/mod.rs#L18-L50
It is used for light client purposes.

It is quite complicated and should be properly specified.

Additionally, it seems that `ext_storage_changes_root` is supposed to return `None` when this mechanism is disabled.

**Comment by tomaka** [comment]
If any of the implementers finds this issue:

> when this mechanism is disabled.

If I'm not mistaken this is the case on Polkadot and Kusama at the moment, and you must therefore just return `None` from `ext_storage_changes_root`.

**Comment by tomaka** [comment]
Oops, closing as duplicate of https://github.com/w3f/polkadot-spec/issues/129
